---
title: "Import and visualization of ProteinGym data"
author: 
    - name: Tram Nguyen
      affiliation: Department of Biomedical Informatics, Harvard Medical School
      email: Tram_Nguyen@hms.harvard.edu
    - name: Pascal Notin
      affiliation: Department of Systems Biology, Harvard Medical School
    - name: Aaron W Kollasch
      affiliation: Department of Systems Biology, Harvard Medical School
    - name: Ludwig Geistlinger
      affiliation: Department of Biomedical Informatics, Harvard Medical School
package: ProteinGymR
output:
    BiocStyle::html_document:
      self_contained: yes 
      toc: true
      toc_float: true
      toc_depth: 2
      code_folding: show
date: "`r doc_date()`"
vignette: >
    %\VignetteIndexEntry{Data access and visualization}
    %\VignetteEncoding{UTF-8}
    %\VignetteEngine{knitr::rmarkdown}
editor_options: 
    markdown: 
      wrap: 80
---

```{r setup, include = FALSE}
knitr::opts_chunk$set(
    collapse = TRUE,
    comment = "#>",
    crop = NULL,
    message = FALSE
)
```

# Installation

Install the package using Bioconductor. Start R and enter:

```{r, eval = FALSE}
if(!requireNamespace("BiocManager", quietly = TRUE))
        install.packages("BiocManager")
BiocManager::install("ProteinGymR")
```

# Setup

Now, load the package and dependencies used in the vignette.

```{r, message = FALSE}
library(ProteinGymR)
library(tidyr)
library(dplyr)
library(stringr)
library(ggplot2)
library(ComplexHeatmap)
```

# Introduction

Predicting the effects of mutations in proteins is critical to many
applications, from understanding genetic disease to designing novel proteins
to address our most pressing challenges in climate, agriculture and
healthcare. Despite an increase in machine learning-based protein modeling
methods, assessing the effectiveness of these models is problematic due to
the use of distinct, often contrived, experimental datasets and variable
performance across different protein families.

ProteinGym v1.0 is a large-scale and holistic set of benchmarks specifically
designed for protein fitness prediction and design. It encompasses both a
broad collection of over 250 standardized deep mutational scanning assays,
spanning millions of mutated sequences, as well as curated clinical datasets
providing high-quality expert annotations about mutation effects.
Furthermore, ProteinGym reports the performance of a diverse set of over 60
high-performing models from various subfields (eg., mutation effects,
inverse folding) into a unified benchmark.


# Data

The `ProteinGymR` package provides analysis-ready datasets from ProteinGym 
(Notin et. al.,
[2023](https://papers.nips.cc/paper_files/paper/2023/hash/cac723e5ff29f65e3fcbb0739ae91bee-Abstract-Datasets_and_Benchmarks.html)).

1.  Deep mutational scanning assay (DMS) scores from 216 assays measuring
the impact of all possible amino acid substitutions across 186 proteins. The
data is provided with `dms_substitutions()`.

2.  AlphaMissense pathogenicity scores for ~1.6 M substitutions in the
ProteinGym DMS data. The data is provided with `am_scores()`.

3. Five model performance metrics ("AUC", "MCC", "NDCG", "Spearman", 
"Top_recall") for 62 models across 217 assays calculated on DMS substitutions 
in a zero-shot setting. The data is provided with `zeroshot_DMS_metrics()`.


# Explore and visualize data

This vignette explores and visualizes the first dataset of DMS scores. 

Deep mutational scanning is an experimental technique that provides 
comprehensive data on the functional effects of all possible single mutations 
in a protein. For each position in a protein, the amino acid residue is mutated 
and the fitness effects are recorded. While most mutations tend to be 
deleterious, some can enhance protein activity. In addition to analyzing single 
mutations, this method can also examine the effects of multiple mutations, 
yielding insights into protein structure and function. Overall, DMS scores 
provide a detailed map of how changes in a protein's sequence affect its 
function, offering valuable yet complex insights for researchers studying 
protein biology.


## Load and explore the DMS data from ExperimentHub

Datasets in `ProteinGymR` can be easily loaded with built-in functions.

```{r import dms}
dms_data <- dms_substitutions()
```

View the DMS study names for the first 6 assays.
```{r view studies}
head(names(dms_data))
```

View an example of one DMS assay.
```{r view assay}
head(dms_data[[1]])
```

For each DMS assay, the columns show the UniProt protein identifier, the DMS 
experiment assay identifier, the mutant at a given protein position, the mutated
protein sequence, the recorded DMS score, and a binary DMS score bin 
categorizing whether the mutation has an affect on fitness (1) or not (0). For 
more details, access the function documentation with `?dms_substitutions()` and
the reference publication from Notin et al. 2023.


## Visualization of DMS data with ComplexHeatmap

Explore an assay and create a heatmap of the DMS scores.
```{r ACE2}
ACE2 <- dms_data[["ACE2_HUMAN_Chan_2020"]]
```

We want to grab the reference amino acid, protein position, and mutant residue 
from the "mutant" column of the dataset.

```{r ACE2 split mutant}
ACE2 <- 
    ACE2 |>
    dplyr::mutate(
        ref = str_sub(ACE2$mutant, 1, 1),
        pos = as.integer(
            gsub(".*?([0-9]+).*", "\\1", ACE2$mutant)
            ),
        alt = str_sub(ACE2$mutant, -1)
    )

ACE2 <- ACE2 |> select("ref", "pos", "alt", "DMS_score")

head(ACE2)
```

```{r reshape ACE2}
## Reshape the data to wide format
ACE2_wide <- ACE2 |> 
    select(-ref) |> 
    pivot_wider(names_from = alt, values_from = DMS_score) |> 
    arrange(pos)

## Subset to first 100 position
ACE2_wide <- ACE2_wide |> 
    filter(pos <= 100)

head(ACE2_wide)
```

```{r as matrix}
## Convert to matrix
pos <- ACE2_wide$pos
alt <- colnames(ACE2_wide)
alt <- alt[-c(1)]

heatmap_matrix <- ACE2_wide |> 
    select(2:length(ACE2_wide)) |> 
    as.matrix()

## Set amino acid position as rownames of matrix
rownames(heatmap_matrix) <- pos

## Transpose so position is x-axis
heatmap_matrix <- t(heatmap_matrix)
```

```{r ACE2 heatmap, warning=FALSE, fig.wide = TRUE}
## Create the heatmap
ComplexHeatmap::Heatmap(heatmap_matrix,
        name = "DMS Score",
        cluster_rows = FALSE,
        cluster_columns = FALSE,
        show_row_names = TRUE,
        show_column_names = TRUE)
```

The heatmap shows the DMS score at each position along the given protein
(x-axis) where a residue was mutated (alternate amino acid on y-axis). For this 
demonstration, we subset to the first 100 positions. 

A low DMS score indicates low fitness, while a higher DMS score indicates high 
fitness.

Based on the "ACE2_HUMAN_Chan_2020" assay, we can see that at positions 90 and 
92, fitness remained high despite across amino acid changes; possibly 
suggestive of a benign region of the protein. However, several mutations at 
position 48 resulted in low fitness. This could represent an important region 
for protein function where any perturbation would likely be deleterious.


# Benchmarking across models

We will now use the built-in function `benchmark_models()` to compare 
performance across several variant effect prediction models calculated on the 
217 DMS assays in the zero-shot setting. This function takes in one of the five 
available metrics, and compares up to 5 models of the 62 available.

To view all available models, use the function: `available_models()`

```{r, available_models}
available_models()
```

Plot the AUC metric for 5 models.

```{r, warning=FALSE, fig.wide = TRUE}
benchmark_models(metric = "AUC", 
    models = c("GEMME", "CARP_600K", "ESM_1b", "EVmutation", "ProtGPT2"))
```

Based on the AUC metric of evaluation, GEMME performed the best while of the 5 
selected models. If the `metric` argument is not defined, the default used is 
a Spearman correlation. For more information about the models and metrics, see 
the function documentation `?benchmark_models()`.


# References

Notin, P., Kollasch, A., Ritter, D., van Niekerk, L., Paul, S., Spinner, H., 
Rollins, N., Shaw, A., Orenbuch, R., Weitzman, R., Frazer, J., Dias, M., 
Franceschi, D., Gal, Y., & Marks, D. (2023). ProteinGym: Large-Scale 
Benchmarks for Protein Fitness Prediction and Design. In A. Oh, T. Neumann, 
A. Globerson, K. Saenko, M. Hardt, & S. Levine (Eds.), Advances in Neural 
Information Processing Systems (Vol. 36, pp. 64331-64379). 
Curran Associates, Inc.

# Session Info

```{r, sesh info}
sessionInfo()
```
